cmake/OpenCVConfig.cmake

```diff
@@ -47,7 +47,7 @@ endif()
 
 if(NOT DEFINED OpenCV_STATIC)
   # look for global setting
-  if(NOT DEFINED BUILD_SHARED_LIBS OR BUILD_SHARED_LIBS)
+  if(BUILD_SHARED_LIBS)
```

Comment:
I removed the commit that contained the change and force pushed to update
this PR. Now the x64 Linux build fails with an error that does not seem to
be related to the actual code. Can you help with that?